docs/authoring.md

```diff
@@ -259,3 +224,112 @@ There are some template parameters that have a special meaning:
 |Name|Description|
 |---|---|
 |`server`| It is used to let the template know which server from the AsyncAPI specification file you want to use. In some cases, this may be required. For instance, when generating code that connects to a specific server. Use this parameter in case your template relies on users' information about what server from the specification file they want to use during generation. You also need this parameter if you want to use `server.protocol` notation within `conditionalFiles` configuration option. Once you decide to specify this parameter for your template, it is recommended you make it a mandatory parameter otherwise feature like `conditionalFiles` is not going to work if your users do not use this parameter obligatory.
+
+
+## React
+[React](https://reactjs.org) is the render engine which we strongly suggest you should use for any new templates. The only reason it is not the default render engine is to stay backward compatibility.
+* It enables the possiblity of [debugging](#debugging-react-template) your template (this is not possible with Nunjucks).
+* It provides better error stack traces once something is not working.
+* Provides better support for separating code into more manageable chunks/components.
+* The readability of the template are much better then Nunjucks syntax.
+* Better tool support for development.
+
+When writing react templates you decide whether to use CommonJS, ES5 or ES6 modules since everything is bundles together before the rendering process takes over. We use our own React renderer which can be found in the [Generator React SDK](https://github.com/asyncapi/generator-react-sdk). 
```

Comment:
```suggestion
When writing React templates you decide whether to use CommonJS, ES5, or ES6 modules since everything is bundled together before the rendering process takes over. We use our own React renderer which can be found in the [Generator React SDK](https://github.com/asyncapi/generator-react-sdk). 
```